src/handlers.js

```diff
@@ -79,7 +79,13 @@ async function encryptedScanReportHandler(req, res, next) {
 async function scanReportHandler(req, res, next, file) {
     const config = getConfig();
     const { domain, mediaId } = req.params;
-    const { clean, info } = await getReport(req.console, domain, mediaId, file, config.scan);
+    let result = await getReport(req.console, domain, mediaId, file, config.scan);
+
+    if (!result.scanned) {
+       result = await generateReport(req.console, domain, mediaId, file, config.scan);
+    }
+
```

Comment:
It seems odd that this is having to worry about checking the cache itself and calling another function if the cache missed.